NEWS.md

```diff
@@ -1,3 +1,14 @@
+# growthcleanr 2.0.3 - 2022-11-01
+
+## Added
+
+- CRAN release checklist now added under Developer Guidelines vignette
```

Comment:
Could add ticket references here, e.g. `(#99)` at the end. Not critical, though.